frontend/src/metabase/admin/settings/components/widgets/SettingInput/SettingInput.tsx

```diff
@@ -10,33 +10,39 @@ const getValue = (value: string, type: string) => {
   return value;
 };
 
-interface SettingInputProps {
+type Value = string | number | null;
+
+export interface SettingInputProps {
   setting: {
     key: string;
     value: string | null;
     default?: string;
     placeholder?: string;
   };
-  onChange: (value: string | number | null) => void;
+  onChange: (value: Value) => void;
   autoFocus?: boolean;
   fireOnChange?: boolean;
   errorMessage?: string;
   id?: string;
   type?: string;
+  normalize?: (value: Value) => Value;
 }
 
-const SettingInput = ({
+const identity = (value: Value) => value;
```

Comment:
Previously I was inlining this in the component prop, but that will create a new function every time, so it was incorrect before. This change ensures that every render, the `normalize` would be the same across renders if caller doesn't pass the `normalize` prop